### DISCUSSION TITLE
Resistive Patches

### DISCUSSION THREAD

**Discussion by JohannesSchwaiger** [WPI-AIM/ambf#169] [Q&A]
Hey Adnan,

thanks in advance for your time. 

At the moment I am trying to create a resistive gripper to measure interaction forces between a gripper and an object. 

I found that the resistive grippers from your examples don't seem to work on my end as they do in your dissertation. Most gripper don't show the sensors (green and red balls) and using the ones that do(the box and the pr2 gripper resistive), do not really feel resistive. 

I would like to add a resistive patch to an existing gripper model I will attach [here (I use the mica_gripper_upscaled.yaml file)](https://github.com/WPI-AIM/ambf/files/8154678/gripper_resistive.zip). I tried adding the assumed relevant parts, but it does not work yet. I want to grab a small [capsule (capsule_primitive.yaml)](https://github.com/WPI-AIM/ambf/files/8154709/capsule.zip) to start with.

Does the blender addon have options for resistive patches? 

Also, how do I read forces on the objects generated during contact? In the standard object state wrenches or is there a separate ros-topic I didn't find?

I also tried it without resistive patches and things usually slip out of the gripper and the forces are very unstable.  Can something be done there?

Thanks again for your work.

Kind regards,
Johannes

**Comment by adnanmunawar** [comment]
Hi Johannes,

> At the moment I am trying to create a resistive gripper to measure interaction forces between a gripper and an object.
I found that the resistive grippers from your examples don't seem to work on my end as they do in your dissertation. Most gripper don't show the sensors (green and red balls) and using the ones that do(the box and the pr2 gripper resistive), do not really feel resistive.

That piece of code has not been touched in a while and its possible that the migration to `ambf-1.0` and then `ambf-2.0` may have broken some of it. When you say it doesn't work, can you post a short video (gif) of what is happening?

> Does the blender addon have options for resistive patches?
It doesn't unfortunately, you had to add it to the ADFs directly.

>Also, how do I read forces on the objects generated during contact? In the standard object state wrenches or is there a separate ros-topic I didn't find?

Those forces are not reported via the ROS communication. One will have to add specific code to publish this data.

> I also tried it without resistive patches and things usually slip out of the gripper and the forces are very unstable. Can something be done there?

I am not sure what you application is, but if you simply want to grasp an object, and make sure it does not slip out, you can use a proximity sensor to detect an object, and an actuator to constrain it. Both the sensor and the actuator can be parented a rigid body, which may be a link in your gripper. We use this approach in the surgical robotics challenge assets. Please check here:

https://github.com/collaborative-robotics/surgical_robotics_challenge/blob/master/scripts/surgical_robotics_challenge/psm_arm.py#L139

and here:

https://github.com/collaborative-robotics/surgical_robotics_challenge/blob/master/ADF/sensors_actuators_psm1.yaml
